asset/records.go

```diff
@@ -15,12 +14,7 @@ import (
 type LeafTlvType = tlv.Type
 
 const (
-	LeafVersion LeafTlvType = 0
-	// TODO: LeafID and LeafGenesis have conflicting types. The BIP
-	// specifies LeafID, but this would require wallets to store the
-	// preimage to the asset ID elsewhere, when it could be included in the
-	// TLV itself as done with LeafGenesis.
-	LeafID                  LeafTlvType = 1
+	LeafVersion             LeafTlvType = 0
```

Comment:
`genesis` is replaced with pure `assetID` in the BIP, but we can make that change in a follow-up PR.